File: src/main/scala/com/exasol/cloudetl/transform/Transformation.scala
package com.exasol.cloudetl.transform

/**
 * An interface for implementing data transformations before importing into a database or exporting to storage.
 */
trait Transformation {

  /**
   * Applies transformations to the list of objects.
   *
   * @param values a row of objects
   * @return tranformed list of objects
   */
  def transform(values: Array[Object]): Array[Object]
}

File: src/main/java/com/exasol/cloudetl/transform/Transformation.java
package com.exasol.cloudetl.transform;

/**
 * An interface for implementing data transformations before importing into a
 * database or exporting to storage.
 */
public interface Transformation {
    /**
     * Applies transformations to the list of objects.
     *
     * @param values row of objects
     * @return transformed list of objects
     */
    Object[] transform(Object[] values);
}
